METRICS.md

```diff
@@ -0,0 +1,91 @@
+# Near Metrics Instructions
+
+## Setup Prometheus
+
+Download a copy of Prometheus from https://prometheus.io/ selecting the OS and extracting the relevant files.
+
+Alternatively, you can also use your favorite package manager. For example, in Arch linux:
+
+`$ sudo pacman -S prometheus`
+
+Or on Mac OS:
+
+`$ brew install prometheus`
+
+## Running
+
+First run Prometheus, using the `prometheus-near.yml` config file located in the `./conf` directory of this repository:
+
+`$ prometheus --config.file=$PATH_TO_REPO/conf/prometheus-near.yml`
+
+Build the NEARCore node:
+
+`$ cargo build -p near`
+
+Generate the initial configuration (genesis state, validator keys and node keys):
+
+`$ $PATH_TO_REPO/target/debug/near init`
+`$ $PATH_TO_REPO/target/debug/near testnet`
+
+Then run a NEAR node0:
+
+`$ $PATH_TO_REPO/target/debug/near --home ~/.near/node0 run`
+
+Extract the server details from the NEAR boot node, it will look similar to:
+
+`ed25519:BT6VhbcNeuq9KXUYpELanWMnSho2LFC5EiGP2Wx7S7gj@0.0.0.0:24567`
+
+Run nodes 1 to 3, replacing `--boot-nodes` with the details output from `node0`
+and X with 1, 2 then 3 (executing the addition symbols).
+
+```
+target/release/near --home ~/.near/nodeX run \
+                    --boot-nodes "ed25519:BT6VhbcNeuq9KXUYpELanWMnSho2LFC5EiGP2Wx7S7gj@0.0.0.0:24567" \
+                    --network-addr 0.0.0.0:(24567 + X) --rpc-addr 0.0.0.0:(3031 + X)
+```
+
+## Setup and run Grafana
+
+Download a copy of Grafana from https://grafana.com/ selecting the OS and extracting the relevant files.
+
+Alternatively, you can also use your favorite package manager. For example, in Arch linux:
+
+`$ sudo pacman -S grafana`
+
+Start grafana as a service:
+
+`$ systemctl start grafana`
+
+Browse to the following address: `localhost:3000` (default username/password is `admin/admin`):
+
+Click `Create your first datasource` -> `Prometheus` and update the following fields:
+
+`Name`: Prometheus
+`URL`: http://localhost:9090
```

Comment:
i think this is the port to be exported. A question is on deployment, should prometheus lives in same instance as nearcore? It could be:
```
+------------------------+
| nearcore + prometheus  |--
+------------------------+  \-
                              \--
+------------------------+       \-  +-----------+
| nearcore + prometheus  |----------/+ grafana   |
+------------------------+       /-- +-----------+
                               /-
+------------------------+  /--
| nearcore + prometheus  |--
+------------------------+
```
Or:
```
+-------------+                                    
| nearcore    +---                                 
+-------------+   \-------                         
+-------------+           \--+--------------+      +-------------+
| nearcore    |--------------+-rometheus    +------+ grafana     |
+-------------+           /--+--------------+      +-------------+
+-------------+   /-------
| nearcore    +---
+-------------+
```
I would prefer the later since it can separately scale prometheus, And I think metric collection happens less often